docs/core/advanced_tutorials/using-results.md

```diff
@@ -0,0 +1,226 @@
+
+```python
+# flow.py
+from prefect.engine.results import GCSResult
+
+gcs_result = GCSResult(bucket='prefect_results', location='my_task.txt')
+
+@task(result=gcs_result)
+def my_task():
+    # ...
+```
+
+Now my flow will always store this task's data at `gs://prefect_results/my_task.txt` directly.
+
+## Templating a task's location
+
+You can also provide a templatable string to a result subclass' `location` attribute. You can include in the template string anything that is in `prefect.context`, including any custom context you set.
+
+For example, we can achieve the same thing as our prior example but using the fact that `prefect.context.task_name` will resolve to the name task's function at runtime:
+
+```python
+# flow.py
+from prefect.engine.results import GCSResult
+
+gcs_result = GCSResult(bucket='prefect_results', location='{task_name}.txt')
+
+@task(result=gcs_result)
+def my_task():
+    # ...
+```
+This will also store my task's data at `gs://prefect_results/my_task.txt`, but it interpolated the task's name from context instead of requiring it to be hardcoded in the result initialization.
+
+This can become quite powerful and allow you to reuse the same result instance to configure multiple tasks to be persisted, based on distinguishing contextual information for each task. It becomes even easier if we configure the whole flow at once by providing a result that has a `location` attribute that, when formatted, will be disambiguous for each task. For example, the following flow will persist the return value for `my_task` and `my_second_task` in separate GCS keys named after each task, with limited code configuration.
+
+```python{4,14}
+# flow.py
+from prefect.engine.results import GCSResult
+
+gcs_result = GCSResult(bucket='prefect_results', location='{task_name}.txt')
+
+@task
+def my_task():
+    # ...
+
+@task
+def my_second_task():
+    # ...
+
+with Flow("my handled flow!", result=gcs_result):
+    my_task()
+    my_second_task()
+```
+
+```bash
+$ gsutil ls -r gs://prefect_results
+gs://prefect_results/my_task.txt
+gs://prefect_results/my_second_task.txt
+```
+
+## Specifying a location for mapped or looped tasks
+
+Mapped or looped tasks require special consideration when it comes to their result location, as they may run many times and each of their results should be stored separately.
+
+## Running a flow with `PrefectResult`
+
+A unique case is the `PrefectResult`, as it serializes its `PrefectResult.value` as its `Result.location`. This is only useful for small data loads and for data that Cloud users are comfortable sharing with the Cloud database. When used effectively, they can provide efficient inspection of data output in the UI. For this reason, all `Parameter` type Tasks use this as their result subclass.
+
+Let's see this with an example. The same flow that adds numbers together will instead be configured with the `PrefectResult`:
+
+```python
+# flow.py
+from prefect.engine.result import PrefectResult
```

Comment:
```suggestion
from prefect.engine.results import PrefectResult
```

Right now it's a bit confusing - we have two submodules (result and results); I think the goal should be to migrate entirely to results (plural) as there are no public classes in `result`